pkg/server/plugin/upstreamauthority/disk/disk.go

```diff
@@ -102,13 +126,22 @@ func (p *Plugin) Configure(_ context.Context, req *configv1.ConfigureRequest) (*
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
 
-	p.config = config
+	p.config = newConfig
 	p.certs = certs
 	p.upstreamCA = upstreamCA
 
 	return &configv1.ConfigureResponse{}, nil
 }
 
+func (p *Plugin) Validate(_ context.Context, req *configv1.ValidateRequest) (*configv1.ValidateResponse, error) {
+	_, notes, err := NewConfiguration(req.HclConfiguration, req.CoreConfiguration)
+
+	return &configv1.ValidateResponse{
```

Comment:
I think that we should probably check for an error here. If there is an error, I would expect the ValidateResponse to be nil.

Reply:
The error is checked and sent in the response.  Whether a validation succeeds or fails, the response is always sent successfully, because the response contains whether it succeeded or failed.

That was the design that was agreed upon by the maintainers.